pkg/alertmanager/statefulset.go

```diff
@@ -624,6 +627,33 @@ func makeStatefulSetSpec(logger *slog.Logger, a *monitoringv1.Alertmanager, conf
 		configReloaderVolumeMounts = append(configReloaderVolumeMounts, configMount...)
 	}
 
+	if !version.LT(semver.MustParse("0.24.0")) {
+		var fields monitoringv1.ClusterTLSConfigFields
+
+		if a.Spec.ClusterTLSConfig != nil {
+			fields = *a.Spec.ClusterTLSConfig
+		}
+
+		clusterTLSConfig, err := clustertlsconfig.New(clusterTLSConfigDir, clusterTLSConfigSecretName(a.Name), fields)
+		if err != nil {
+			return nil, err
+		}
+
+		confArg, configVol, configMount, err := clusterTLSConfig.GetMountParameters()
+		if err != nil {
+			return nil, err
+		}
+
+		volumes = append(volumes, configVol...)
+		amVolumeMounts = append(amVolumeMounts, configMount...)
+
+		// TODO: We need to validate ClusterTLSConfig or we run the risk of ClusterTLSConfig != nil but ServerTLS == nil
```

Comment:
Not sure to grasp this comment. Is there an edge case where the operator might generate an invalid deployment?

Reply:
I have to be honest, I don't remember what the comment stood for. I will have to let you know later :thinking: